src/resources/js/admin/order-modifiers/amount-field.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * File: coupons.js
+ *
+ * Dependencies:
+ * - jQuery
+ * - window.tribe.validation
+ * - window.etOrderModifiersAmountField
+ */
+
+// Fallbacks for the global variables.
+window.etOrderModifiersAmountField = window.etOrderModifiersAmountField || {
+	currencySymbol: '$',
+	decimalSeparator: '.',
+	placement: 'before',
+	precision: 2,
+};
+
+( function( $, validation, i18n ) {
+	const $document = $( document );
+
+	const selectors = {
+		amount: '#order_modifier_amount',
+		save: '#order_modifier_form_save',
+		type: '#order_modifier_sub_type',
+	};
+
+	const getType = () => {
+		const $typeSelect = $( selectors.type );
+		return $typeSelect.val();
+	};
+
+	const addAmountExtras = () => {
+		const $amount = $( selectors.amount );
+		let value = Number.parseFloat( $amount.val() || 0 ).toFixed( i18n.precision );
+
+		// If the decimal separator is not a period, replace it.
+		if ( '.' !== i18n.decimalSeparator ) {
```

Comment:
How does this handle currencies where the comma and period are swapped?
i.e. $ 1.234,56 pesos in Argentina

Reply:
The comma would be stripped out in this field.

Reply:
@JPry what stephen is looking for is whether this field respects currency formatting or not?. Like in the example shown `.` is used for thousand separator and `,` is used for decimal separators.

Reply:
Right - so would we get `$ 1.234.56`? That's not all right...
Might be OK if we got `$ 1234.56` and it was then reformatted, but then we're just adding more steps.

Reply:
Honestly, after watching the video I am unsure this is "better" as the value format keeps switching on save/entry.

A better solution might be an input mask or an [input prefix/suffix](https://designsystem.digital.gov/components/input-prefix-suffix/)?

Reply:
In the end, check with design. 😉